### PR TITLE
Fixing crash(es)

### LINKED ISSUE #3: App crashes while selecting gif from gif selector

### PR DESCRIPTION
**Pull request by narendrakpatel**
fixes #3 
:bug:  fixed crash on opening app without permissions 
:bug:  fixed crash on opening gif
:art:  slight UI improvement

### CODE REVIEW

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/BaseActivity.java [hunk 1]
```diff
@@ -106,11 +106,14 @@ protected void onCreate(Bundle savedInstanceState) {
     public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
         super.onRequestPermissionsResult(requestCode, permissions, grantResults);
 
+        Log.d("lololo", "took permissions");
```
Comment (Mansi449) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/BaseActivity.java]: Define and use a proper LOG_TAG here. See this https://developer.android.com/reference/android/util/Log.html

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/BaseActivity.java [hunk 2]
```diff
@@ -106,11 +106,14 @@ protected void onCreate(Bundle savedInstanceState) {
...
         super.onRequestPermissionsResult(requestCode, permissions, grantResults);
 
+        Log.d("lololo", "took permissions");
+
         boolean allGranted = true;
 
         if (grantResults.length > 0) {
             for (int grantResult : grantResults) {
                 if (grantResult != PackageManager.PERMISSION_GRANTED) {
+                    Log.d("lololo", String.valueOf(grantResult));
```
Comment (Mansi449) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/BaseActivity.java]: Use LOG_TAG

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/ContactListFragment.java [hunk 3]
```diff
@@ -1,17 +1,22 @@
 package com.sdsmdg.pulkit.callingtext;
 
+import android.*;
```
Comment (Mansi449) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/ContactListFragment.java]: This import statement is not needed. Remove it.

File: app/src/main/java/com/sdsmdg/pulkit/callingtext/ContactListFragment.java [hunk 4]
```diff
@@ -1,17 +1,22 @@
...
+import android.content.pm.PackageManager;
 import android.database.Cursor;
 import android.os.AsyncTask;
 import android.os.Bundle;
 import android.provider.ContactsContract;
 import android.support.design.widget.TabLayout;
+import android.support.v4.app.ActivityCompat;
 import android.support.v4.app.Fragment;
 import android.support.v4.app.LoaderManager;
+import android.support.v4.content.ContextCompat;
```
Comment (Mansi449) [inline app/src/main/java/com/sdsmdg/pulkit/callingtext/ContactListFragment.java]: Not needed

**Review by Mansi449 — CHANGES_REQUESTED** [review]
Good work! But do make the requested changes.

File: build.gradle [hunk 5]
```diff
@@ -3,6 +3,7 @@
 buildscript {
     repositories {
         jcenter()
+        google()
     }
     dependencies {
```
Comment (Mansi449) [inline build.gradle]: Why these blank lines

File: build.gradle [hunk 6]
```diff
@@ -24,6 +25,7 @@ allprojects {
             url "https://jitpack.io"
         }
```
Comment (Mansi449) [inline build.gradle]: remove blank line

**Review by Mansi449 — CHANGES_REQUESTED** [review]

**Comment by narendrakpatel** [comment]
I have done the changes.
Are there any other changes that i should do?

**Review by Mansi449 — APPROVED** [review]

**Comment by Mansi449** [comment]
No, it's okay for now.

**Pull request merged**